### DISCUSSION TITLE
Recording and reviewing URL replacements

### DISCUSSION THREAD

**Discussion by xhluca** [ServiceNow/BrowserGym#246] [Q&A]
This code is used to replace the pattern `__SHOPPING__` into the correct URL. Is there a way to inspect the replacement post-hoc (after inference) to ensure that tasks were correctly inspected?

Here's the replacement code:

https://github.com/ServiceNow/BrowserGym/blob/66cf40215dd12269bfcea3e22e514b804f09a9b2/browsergym/webarena/src/browsergym/webarena/task.py#L56-L63

I have noticed that in many cases a reward of 0 is assigned even though the correct URL was navigated at certain point

**Comment by gasse** [comment]
Hi @xhluca , I don't think you will be able to see that post-hoc in the traces. The easiest way to check would be to set yourself in debug mode, place a few breakpoints in the code and inspect the self.config variable. You could also go down to webarena's evaluator function and try to understand at which point the URLs are checked.
